Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -156,6 +177,41 @@ fileprivate class ParameterReplacementVisitor: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  override func visit(_ node: GenericArgumentClauseSyntax) -> SyntaxVisitorContinueKind {
+    return .visitChildren
+  }
+
+  override func visit(_ node: GenericArgumentListSyntax) -> SyntaxVisitorContinueKind {
+    return .visitChildren
+  }
+
+  override func visit(_ node: GenericArgumentSyntax) -> SyntaxVisitorContinueKind {
+    let baseName = node.argument
+    guard let genericParameterClause = macro.genericParameterClause else {
+      return .skipChildren
+    }
+
+    let matchedParameter = genericParameterClause.parameters.enumerated().first { (index, parameter) in
+      return parameter.name.text == "\(baseName)"
+    }
```

Comment:
You should be able to use `firstIndex(where:)`

```suggestion
    let parameterIndex = genericParameterClause.parameters.firstIndex { (index, parameter) in
      return parameter.name.text == "\(baseName)"
    }
```

Reply:
I think you’re still using `first` and then only using the index. Is there any reason why `firstIndex` doesn’t work?

Reply:
Using firstIndex I run into: 

```
/Users/ktoso/code/swift-project/swift-syntax/Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift:215:71: error: cannot convert value of type 'SyntaxChildrenIndex' to expected argument type 'Int'
    genericReplacements.append(.init(reference: node, parameterIndex: parameterIndex))
                                                                      ^
```

so kept first and getting the index integer; is there a better trick here?

Reply:
For what it's worth other code uses the same pattern as we have here.